prov/shm/src/smr_msg.c

```diff
@@ -209,11 +217,15 @@ static ssize_t smr_generic_sendmsg(struct smr_ep *ep, const struct iovec *iov,
 		}
 		resp = ofi_cirque_next(smr_resp_queue(ep->region));
 		pend = ofi_freestack_pop(ep->pend_fs);
+		pend->has_fd = false;
 		if (smr_cma_enabled(ep, peer_smr) && iface == FI_HMEM_SYSTEM) {
 			smr_format_iov(cmd, iov, iov_count, total_len, ep->region,
 				       resp);
 		} else {
-			if (iface == FI_HMEM_ZE &&
+			if (should_use_ipc) {
+				ret = smr_format_ipc(cmd, iov, total_len,
+						     ep->region, resp, iface);
+			} else if (iface == FI_HMEM_ZE &&
```

Comment:
This if/else needs to be reworked - ZE will always go to the first case if IPC is enabled (assuming the p2p call gets merged with the IPC enabled call as referenced in a separate comment) and never to the second. We should also be able to rework some of the ZE conditions as well. iov_count can go for sure. ze_ipc_enabled should stay since that has to do with the ZE specific sockets. The other two should probably get added to the use_ipc condition since I think the same conditions apply for both CUDA and ROCR.